src/ColonesAssetAdapter.ts

```diff
@@ -0,0 +1,145 @@
+import { Htlc as OasisHtlc, HtlcStatus, SettlementInstruction, SettlementStatus } from '@nimiq/oasis-api';
+import { AssetAdapter, SwapAsset } from './IAssetAdapter';
+
+export type HtlcDetails = OasisHtlc;
+
+export interface OasisClient {
+  getHtlc(id: string): Promise<HtlcDetails>;
+  settleHtlc(id: string, secret: string, settlementJWS: string, authorizationToken?: string): Promise<HtlcDetails>;
+}
```

Comment:
Since this is the same interface as for the Euro adapter, can we share it between the two? And I guess the `HtlcDetails` type above, too. Maybe these two can be exported from a common `OasisCommon.ts` file?

Reply:
Yeah, I didn't want to refactor that much, but it is way better. I will update it